### PR TITLE
 Explicitly encode URI in test for sinatra upgrade

### PR DESCRIPTION
**Pull request by Hamms**
# Description

The [sinatra upgrade](#25507) introduced an unknown change that resulted in this test throwing a misleading error.

Specifically, the test began failing with the message:

    Minitest::Assertion: unexpected invocation: #<AnyInstance:Firebase::Client>.get('/v3/channels/TP3c4-GKEI4pvQ5HayE-1w-test-suffix/storage/tables/my table/records')
    unsatisfied expectations:
    - expected exactly once, not yet invoked: #<AnyInstance:Firebase::Client>.get('/v3/channels/TP3c4-GKEI4pvQ5HayE-1w-test-suffix/storage/tables/my%20table/records')

The issue is simply that the test now expects us to check for the unencoded string rather than the encoded string. It's not clear why this changed, but manual testing confirms that the actual functionality being tested in unchanged.

The simple fix is to continue making the request with a properly-encoded URI, but to check for a decoded URI.

## Links

- [base PR](https://github.com/code-dot-org/code-dot-org/pull/25507)
- [jira](https://codedotorg.atlassian.net/browse/FND-795)

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Review by davidsbailey — APPROVED** [review]
LGTM. thanks for testing manually! that seems convincing that the new test case is correct.

**Review by breville — APPROVED** [review]

File: Gemfile [hunk 1]
```diff
@@ -28,7 +28,7 @@ gem 'sprockets-rails'
 # (see: http://guides.rubyonrails.org/4_2_release_notes.html#respond-with-class-level-respond-to)
 gem 'responders', '~> 2.0'
 
-gem 'sinatra', '~> 2.0.0.beta2', require: 'sinatra/base'
```
Comment (breville) [inline Gemfile:31]: hah, we've been on a beta for the last three years?

**Comment by Hamms** [comment]
Aw, dangit. I got paranoid that I when I tried to test this manually I didn't actually do so on the correct version of sinatra so I decided to test again, and it turns out I was right to be so! This upgrade actually _does_ regress functionality.

@davidsbailey, do you remember why exactly this change https://github.com/code-dot-org/code-dot-org/pull/18010/commits/308e048ceb467e4beae82edf33740b5995ba01a2 did in fact fix things? Based on a little fiddling with the code, it seems like reverting this change is what's necessary to get things to _work_ in sinatra 2.0.2

**Comment by davidsbailey** [comment]
> @davidsbailey, do you remember why exactly this change [308e048](https://github.com/code-dot-org/code-dot-org/commit/308e048ceb467e4beae82edf33740b5995ba01a2) did in fact fix things? Based on a little fiddling with the code, it seems like reverting this change is what's necessary to get things to _work_ in sinatra 2.0.2

I believe this fixed things because we were not properly encoding table names with spaces in them -- we simply never tested that case when we built the initial export feature. I imagine the best path forward would be to debug locally to reach an escaping that does give the right behavior when table names contain spaces, and then update the test (and possibly the client code) to match.

If this turns out to be a stumper, it could also be possible to move this API to dashboard, where we have a better understanding of how to encode and decode strings. 

According to https://github.com/code-dot-org/code-dot-org/pull/9871 , it may also be possible to have the client do the export, if you can find a suitable polyfill for the `download` attribute to make it work in IE11.

**Comment by Hamms** [comment]
Haha oh boy I figured out why my manual test worked previously and not this most recent time.

It turns out, the existing functionality works just fine on sinatra `2.0.2`, which at first glance seems to be the version this change is trying to upgrade to. However, we are actually upgrading [to sinatra `2.0.5`](https://github.com/code-dot-org/code-dot-org/pull/31614/commits/13215d833ff1aef20cea9df02d70058884cb6c2c#diff-e79a60dc6b85309ae70a6ea8261eaf95R781), on which things actually break.

This also explains why I had so much trouble trying to track down the change in sinatra that broke things.

Rather than trying to investigate the nature of the break, I'm just going to pin our upgrade to `2.0.2`. I suggest we schedule some follow-up work to figure out what in `2.0.5` breaks things.

File: Gemfile [hunk 2]
```diff
@@ -28,7 +28,10 @@ gem 'sprockets-rails'
...
+# Pinning sinatra to 2.0.2, since '~> 2.0.2' actually lands us on 2.0.5, which
+# breaks some firebase URIs. See
+# https://github.com/code-dot-org/code-dot-org/pull/31614
+gem 'sinatra', '2.0.2', require: 'sinatra/base'
```
Comment (davidsbailey) [inline Gemfile:34]: whoa, I missed this subtlety. thanks for documenting!

**Review by davidsbailey — APPROVED** [review]

File: Gemfile (hunk 1 above)
Comment (Hamms) [reply Gemfile:31]: yup :upside_down_face:

**Review by wjordan — APPROVED** [review]

**Pull request merged**